test/common_utils.py

```diff
@@ -404,11 +404,15 @@ def assertTensorsEqual(a, b):
                     else:
                         b = b.to(a)
 
-                    if x.dtype == torch.bool and y.dtype == torch.bool:
-                        self.assertEqual(x.tolist(), y.tolist())
-                    elif x.dtype == torch.bool or y.dtype == torch.bool:
+                    if  ((a.dtype != torch.bool and b.dtype == torch.bool) or
+                         (a.dtype == torch.bool and b.dtype != torch.bool)):
                         raise TypeError("Was expecting both tensors to be bool type.")
                     else:
+                        if a.dtype == torch.bool and b.dtype == torch.bool:
+                            self.assertEqual(a.tolist(), b.tolist())
+                            a = torch.as_tensor(a, dtype=torch.int)
+                            b = torch.as_tensor(b, dtype=torch.int)
```

Comment:
This is pretty un-expected.  Why should we convert into an int? Haven't we already tested that the contents of the two tensors are equal?

Reply:
I think this would make sense if you just got rid of the first assertEqual.  It's also clearer if you just used `to` instead of `as_tensor` -- you already know it's a tensor.  And probably want a comment about why you want to do this (because subtraction isn't currently supported on bool).